components/feature/intent/src/main/java/mozilla/components/feature/intent/IntentProcessor.kt

```diff
@@ -39,71 +36,28 @@ class IntentProcessor(
     private val useDefaultHandlers: Boolean = true,
     private val openNewTab: Boolean = true,
     private val isPrivate: Boolean = false
-) {
-    private val defaultActionViewHandler = { intent: Intent ->
-        val safeIntent = SafeIntent(intent)
-        val url = safeIntent.dataString ?: ""
-
-        when {
-            TextUtils.isEmpty(url) -> false
-
-            CustomTabConfig.isCustomTabIntent(safeIntent) -> {
-                val session = Session(url, false, Source.CUSTOM_TAB).apply {
-                    val displayMetrics = context.resources.displayMetrics
-                    this.customTabConfig = CustomTabConfig.createFromIntent(safeIntent, displayMetrics)
-                }
-                sessionManager.add(session)
-                sessionUseCases.loadUrl(url, session, LoadUrlFlags.external())
-                intent.putExtra(ACTIVE_SESSION_ID, session.id)
-                true
-            }
-
-            else -> {
-                val session = createSession(url, private = isPrivate, source = Source.ACTION_VIEW)
-                sessionUseCases.loadUrl(url, session, LoadUrlFlags.external())
-                true
-            }
-        }
-    }
-
-    private val defaultActionSendHandler = { intent: Intent ->
-        val safeIntent = SafeIntent(intent)
-        val extraText = safeIntent.getStringExtra(Intent.EXTRA_TEXT) ?: ""
-
-        when {
-            TextUtils.isEmpty(extraText.trim()) -> false
-
-            else -> {
-                WebURLFinder(extraText).bestWebURL()?.let { url ->
-                    sessionUseCases.loadUrl(
-                        url,
-                        createSession(
-                            url,
-                            private = isPrivate,
-                            source = Source.ACTION_SEND
-                        ),
-                        LoadUrlFlags.external()
-                    )
-                } ?: run {
-                    searchUseCases.newTabSearch(extraText, Source.ACTION_SEND, openNewTab)
-                }
-                true
-            }
-        }
-    }
-
-    private fun createSession(url: String, private: Boolean = false, source: Source): Session {
-        return if (openNewTab) {
-            Session(url, private, source).also { sessionManager.add(it, selected = true) }
-        } else {
-            sessionManager.selectedSession ?: Session(url, private, source)
-        }
-    }
+) : mozilla.components.browser.session.intent.IntentProcessor {
```

Comment:
nit: We probably want to rename the interface or the implementation so they don't have the same name. Maybe the interface could be `IntentProcessHandler` since we have multiple of these handlers?